### ISSUE TITLE
Docking support for controllers

### ISSUE THREAD

**Issue by sacchin-conffx** [ocornut/imgui#5524]
Version/Branch of Dear ImGui:
Version: 1.88
Branch: docking (based on https://github.com/ocornut/imgui/commit/9cd9c2eff99877a3f10a7f9c2a3a5b9c15ea36c6)

Back-end/Renderer/Compiler/OS
Back-ends: Custom
Operating System/Platform: PS4/PS5/Xbox/Windows

My Issue/Question:
Docking is currently only supported on platforms/devices with mouse and/or touch input, but not controllers. We have a use case for docking via controller navigation and wanted to start a discussion on what it would take to add support for it. Any thoughts on this?

**Comment by ocornut** [comment]
I haven’t thought about it. Open to a design proposal of how it would work from user’s point of view.
Technically speaking it is already possible to move and resize windows with gamepad so I would assume this could be made to work along the existing moving operation.

**Comment by sacchin-conffx** [comment]
Yeah we can leverage the existing move logic of the controller. I was thinking of having the "dock point" or "virtual mouse cursor" be the center of the top edge of a window's rect maybe? The docking widgets would show up as normal based on whether this center point overlaps any other dockable window. If a window is docked, then navigating to it and moving it around with Square (or X) + left stick would undock it. It might not be that simple in practice but this could be a good starting point assuming I'm not missing anything else important.
